include/envoy/runtime/runtime.h

```diff
@@ -96,6 +113,12 @@ class Snapshot {
    * @return uint64_t the runtime value or the default value.
    */
   virtual uint64_t getInteger(const std::string& key, uint64_t default_value) const PURE;
+
+  /**
+   * Fetch the raw runtime entries map. The map data is safe only for the lifetime of the Snapshot.
+   * @return std::unordered_map<std::string, const Entry*> the raw map of loaded runtime values.
+   */
+  virtual std::unordered_map<std::string, const Entry*> getAll() const PURE;
```

Comment:
Is the reason you are copying here the const Entry vs. not const Entry? If that's the case, I would probably do two things:
1) I would use std::reference_wrapper to make it clear it can't be null.
2) I would probably return a vector of pairs, vs. a map. I don't think the caller will really ever need the map functionality, is they could otherwise call the snapshot directly if they need a single value.

Reply:
P.S., I'm not sure but you might also be able to just make the internal map store `const Entry`. I'm not sure if this will work or not due to STL copy semantics, but it might. I would try that first.

Reply:
Went with <string, const Entry> much nicer and the getAll can now return a const ref to the map instead. Much better!